paper-typeahead-input.html

```diff
@@ -449,6 +450,14 @@
         // }
       }
     },
+    _candidatesUpdated:function(candidates){
+      if(this.inputValue != undefined){
+        var val = this.inputValue.trim();
+        if(candidates.length > 0 && candidates[0].trim() !== val){
```

Comment:
Okay. Then what about  ["red", "blue", "yellow"] and inputValue is "red" and new list is ["red","white","rose-red"],  in this case if we can the check candidates[0].trim() !== val, then it will not perform the search and it will not get the hit on "rose-red".